--- addons/account/report/account_financial_report.py	2015-03-06 15:04:23.000000000 +0100
+++ addons/account/report/account_financial_report.py	2015-06-01 15:24:45.000000000 +0200
@@ -63,6 +63,7 @@
                 'type': 'report',
                 'level': bool(report.style_overwrite) and report.style_overwrite or report.level,
                 'account_type': report.type =='sum' and 'view' or False, #used to underline the financial report balances
+                'report_id': report.id,
             }
             if data['form']['debit_credit']:
                 vals['debit'] = report.debit
