--- addons/account/account_invoice.py	2015-08-25 02:41:11.000000000 +0200
+++ addons/account/account_invoice.py	2016-01-20 11:55:08.634307155 +0100
@@ -778,6 +778,9 @@
                     am = line2[tmp]['debit'] - line2[tmp]['credit'] + (l['debit'] - l['credit'])
                     line2[tmp]['debit'] = (am > 0) and am or 0.0
                     line2[tmp]['credit'] = (am < 0) and -am or 0.0
+                    qty = l.get('quantity')
+                    if qty:
+                        line2[tmp]['quantity'] = line2[tmp].get('quantity', 0.0) + qty
                     line2[tmp]['tax_amount'] += l['tax_amount']
                     line2[tmp]['analytic_lines'] += l['analytic_lines']
                 else:
